### DISCUSSION TITLE
Is it the best practices that create a custom props interface that extends the BoxProps?

### DISCUSSION THREAD

**Discussion by RekiDunois** [chakra-ui/chakra-ui#5403] [Q&A]
Hi everyone, I have a question here. Say that I have a custom component, which looks like this:

```jsx
export default function MyItem(props:MyItemProps)
{
    return (
            <Box>
               <Heading>{props.context.title}</Heading>
               <Text>{props.context.content}</Text>
            </Box>
    )
}
```

Note that this component has the prop name `context` and the type of props is `MyItemProps`, which could be written like this:

```jsx
export interface MyItemProps
{
   context: ItemContext
}
```

Since this component was wrapped by `Box`, I hope I could use the custom component just like using `Box`, such as:

```jsx
<MyItem bg='pink.400', context={myContext}/>
```

So I think, maybe I can make `MyItemProps` extends from `BoxProps`, so I can have both my custom prop `context` and all the props that could be used by `Box`.

The question is when I import `BoxProps`, the import cost will be large, like 100+k in a single component. I wonder there will be 
any disadvantages from doing this, if so, what is the best practices to combine custom props and chakra component props

**Comment by MuLoo** [comment]
will this help? 
https://chakra-ui.com/community/recipes/as-prop
